lib/internal/test_runner/reporter/dot.js

```diff
@@ -1,18 +1,24 @@
 'use strict';
 
+const colors = require('internal/util/colors');
 const {
   MathMax,
 } = primordials;
 
+const failedTestSymbol = '\u2716 ';
+
 module.exports = async function* dot(source) {
   let count = 0;
   let columns = getLineLength();
-  for await (const { type } of source) {
+  let failedTests = [];
+
+  for await (const { type, data } of source) {
     if (type === 'test:pass') {
       yield '.';
     }
     if (type === 'test:fail') {
       yield 'X';
+      failedTests.push(data);
```

Comment:
For the internals of Node.js, we use primordials, so try to use them when possible (`ArrayPrototypePush`)